src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -545,27 +547,72 @@ Namespace Microsoft.CodeAnalysis.Semantics
             End If
 
             Dim operand As Lazy(Of IOperation)
+            Dim type As ITypeSymbol = boundConversion.Type
+            Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundConversion.ConstantValueOpt)
+            Dim isImplicit As Boolean = boundConversion.WasCompilerGenerated
             Dim methodSymbol As MethodSymbol
 
             If (boundConversion.ConversionKind And VisualBasic.ConversionKind.UserDefined) = VisualBasic.ConversionKind.UserDefined Then
                 Dim userDefinedConversion As BoundUserDefinedConversion = DirectCast(boundConversion.Operand, BoundUserDefinedConversion)
                 methodSymbol = userDefinedConversion.Call.Method
                 operand = New Lazy(Of IOperation)(Function() Create(userDefinedConversion.Operand))
+            ElseIf boundConversion.ConversionKind = VisualBasic.ConversionKind.Identity AndAlso
+                   boundConversion.Operand.Kind = BoundKind.DelegateCreationExpression Then
+                ' In this scenario, we're something like New DelegateType(AddressOf Method)
+                ' We don't want to process both this conversion and the DelegateCreationExpression
+                ' as different delegate creation expressions. So we have the operand here be the
+                ' child of the BoundDelegateCreationExpression
+                methodSymbol = Nothing
+                operand = New Lazy(Of IOperation)(Function() CreateBoundDelegateCreationExpressionChildOperation(DirectCast(boundConversion.Operand, BoundDelegateCreationExpression)))
             Else
                 methodSymbol = Nothing
                 operand = New Lazy(Of IOperation)(Function() Create(boundConversion.Operand))
             End If
 
-            Dim conversion = New Conversion(New KeyValuePair(Of ConversionKind, MethodSymbol)(boundConversion.ConversionKind, methodSymbol))
+            If (boundConversion.Operand.Kind = BoundKind.Lambda OrElse
+                boundConversion.Operand.Kind = BoundKind.QueryLambda OrElse
+                boundConversion.Operand.Kind = BoundKind.RelaxationLambda OrElse
+                boundConversion.Operand.Kind = BoundKind.UnboundLambda OrElse
+                boundConversion.Operand.Kind = BoundKind.DelegateCreationExpression) AndAlso
+               type.InheritsSpecialType(SpecialType.System_Delegate) Then
```

Comment:
>InheritsSpecialType [](start = 20, length = 19)

I believe we already have a helper IsDelegateType

Reply:
Fixed.